cscs-checks/apps/vasp/vasp_check.py

```diff
@@ -5,45 +5,102 @@
 
 import reframe as rfm
 import reframe.utility.sanity as sn
+from hpctestlib.apps.vasp import VASP
 
+dom_cpu = {
+    'maint': (148.7, None, 0.05, 's'),
+    'prod': (148.7, None, 0.05, 's'),
+}
 
-class VASPCheck(rfm.RunOnlyRegressionTest):
-    def __init__(self):
+daint_cpu = {
+    'maint': (105.3, None, 0.20, 's'),
+    'prod': (105.3, None, 0.20, 's'),
+}
+
+eiger_cpu = {
+    'maint': (100.0, None, 0.10, 's'),
+    'prod': (100.0, None, 0.10, 's'),
+}
+
+pilatus_cpu = {
+    'maint': (100.0, None, 0.10, 's'),
+    'prod': (100.0, None, 0.10, 's'),
+}
+
+
+REFERENCE_CPU_PERFORMANCE = {
+    'dom:gpu': dom_cpu,
+    'daint:gpu': daint_cpu,
+    'eiger:mc': eiger_cpu,
+    'pilatus:mc': pilatus_cpu
+}
+
+
+dom_gpu = {
+    'maint': (61.0, None, 0.10, 's'),
+    'prod': (46.7, None, 0.20, 's'),
+}
+
+daint_gpu = {
+    'maint': (46.7, None, 0.20, 's'),
+    'prod': (46.7, None, 0.20, 's'),
+}
+
+REFERENCE_GPU_PERFORMANCE = {
+    'dom:gpu': dom_gpu,
+    'daint:gpu': daint_gpu,
+}
+
+
+class VASPCheck(VASP):
+    modules = ['VASP']
+    maintainers = ['LM']
+    tags = {'scs'}
+    strict_check = False
+    extra_resources = {
+        'switches': {
+            'num_switches': 1
+        }
+    }
+    force_value = -.85026214E+03
```

Comment:
This is input file dependent and, I think they should be moved to the base class. IMHO, the input files should be parameters of the base class.